src/solution.jl

```diff
@@ -193,16 +207,54 @@ DiffEqBase.calculate_solution_errors!(sol::ProbODESolution, args...; kwargs...)
 # Dense Output
 ########################################################################################
 abstract type AbstractODEFilterPosterior <: DiffEqBase.AbstractDiffEqInterpolation end
-struct ODEFilterPosterior <: AbstractODEFilterPosterior end
+struct ODEFilterPosterior{T1,T2,T3,T4,T5,T6} <: AbstractODEFilterPosterior
+    ts::T1
+    x_filt::T2
+    x_smooth::T3
+    diffusions::T4
+    cache::T5
+    smooth::T6
+end
 DiffEqBase.interp_summary(interp::ODEFilterPosterior) =
     "ODE Filter Posterior"
 
-(sol::ProbODESolution)(t::Real, args...) =
-    sol.cache.SolProj * interpolate(
-        t, sol.t, sol.x_filt, sol.x_smooth, sol.diffusions, sol.cache;
-        smoothed=sol.alg.smooth)
-(sol::ProbODESolution)(t::AbstractVector, args...) =
-    DiffEqArray(StructArray(sol.(t, args...)), t)
+function (interp::ODEFilterPosterior)(
+    t::Real,
+    idxs::Nothing,
+    ::Type{deriv},
+    p,
+    continuity,
+) where {deriv}
+    @assert deriv == Val{0} "Derivatives not supported in the interpolation"
+    x = interpolate(
+        t, interp.ts, interp.x_filt, interp.x_smooth, interp.diffusions, interp.cache;
+        smoothed=interp.smooth)
+    return interp.cache.SolProj * x
+end
+function (interp::ODEFilterPosterior)(
+    t::Real,
+    idxs::Integer,
+    ::Type{deriv},
+    p,
```

Comment:
And I'm unsure why there is ever a `p` beingpassed to this? Maybe check in SciMLBase, or ask someone on slack.

Reply:
:shrug: